### ISSUE TITLE
Error when deserializing NamedTuple with multiple union fields

### ISSUE THREAD

**Issue by owen2345** [j8r/crystalizer#12]
I am having error when including array values inside Hashes

- Without array (works well):
```ruby
$> d = {str: "String", number: 10}; seria = Crystalizer::JSON.serialize(d); Crystalizer::JSON.deserialize(seria, typeof(d))
 ==> {str: "String", number: 10}
```
- With array (error when including an array)
```ruby
d = {str: "String", number: 10, arr: [1, "Str", true]}; seria = Crystalizer::JSON.serialize(d); Crystalizer::JSON.deserialize(seria, typeof(d))
```
![image](https://user-images.githubusercontent.com/1148415/114639401-c68e2380-9c9b-11eb-97d3-453dd5b7bc95.png)

**Comment by j8r** [comment]
Related to #3 , but for serializing NamedTuples. Got a fix using a `de_unionize` method to make the dispatch working as expected.

**Comment by j8r** [comment]
Thank you, good catch.

**Comment by owen2345** [comment]
@j8r not clear.
I see that the PR (#4) that includes the fix was already merged into master, but the problem persists.
Could u please share me a sample using `de_unionize `? I see nothing related in the readme (Sorry if my question is too basic, it is my second day with Crystal :D).

**Comment by j8r** [comment]
I said it was related, because it is the same bug class: a method dispatch not properly done by the Crystal compiler.
I pushed a fix, try crystalizer v1.0.0.

**Comment by owen2345** [comment]
It is working now, thank you much!
